examples/direct_fidelity_estimation.py

```diff
@@ -30,16 +30,29 @@
 
 
 def build_circuit():
-    # Builds an arbitrary circuit to test. The circuit is non Clifford to show
-    # the use of simulators.
+    # Builds an arbitrary circuit to test. If the circuit is Clifford, then it
+    # also returns a state from which we can get the stabilitzers. If the
+    # circuit is non Clifford, the state is None.
     qubits = cirq.LineQubit.range(3)
-    circuit = cirq.Circuit(
-        cirq.Z(qubits[0])**0.25,  # T-Gate, non Clifford.
-        cirq.X(qubits[1])**0.123,
-        cirq.X(qubits[2])**0.456)
+    circuit = cirq.Circuit()
+    state = cirq.CliffordState(
+        qubit_map={qubits[i]: i for i in range(len(qubits))})
+    gates = [
+        cirq.Z(qubits[0]),
+        cirq.X(qubits[1]),
+        cirq.X(qubits[2]),
+    ]
+    for gate in gates:
+        circuit.append(gate)
+        if state is not None:
+            try:
+                state.apply_unitary(gate)
+            except ValueError:
+                print('Circuit is non Clifford.')
+                state = None
```

Comment:
`state` being `None` to me seems like a `null` state? Doesn't make sense. Let's change the variable name of `state` to something more fitting. `clifford_state` maybe?

Reply:
Done